File: FinalProject/app/src/main/java/com/example/finalproject/items/Armor.java
package com.example.finalproject.items;


public class Armor extends Item {
    private final int armor, weight, typeOfArmor;

    public Armor(String name, int costSell, int costBuy, int armor, int weight, int typeOfArmor, int category, int rarity) {
        super(name, costSell, costBuy, rarity, category);
        this.armor = armor;
        this.weight = weight;
        this.typeOfArmor = typeOfArmor;
    }

    public int getArmor() {
        return armor;
    }

    public int getWeight() {
        return weight;
    }

    public int getTypeOfArmor() {
        return typeOfArmor;
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/items/Armor.kt
package com.example.finalproject.items

class Armor(name:String, costSell:Double, costBuy:Double,
            val armor:Double, val weight:Double, val typeOfArmor:Int,
            category:Int, rarity:Int)
    :Item(costSell, costBuy, category, rarity, name){
        constructor(armor: Armor):this(armor.name, armor.costSell, armor.costBuy,
            armor.armor, armor.weight,
            armor.typeOfArmor, armor.category, armor.rarity)
    }
